pyomo/common/factory.py

```diff
@@ -0,0 +1,69 @@
+#  ___________________________________________________________________________
+#
+#  Pyomo: Python Optimization Modeling Objects
+#  Copyright 2017 National Technology and Engineering Solutions of Sandia, LLC
+#  Under the terms of Contract DE-NA0003525 with National Technology and
+#  Engineering Solutions of Sandia, LLC, the U.S. Government retains certain
+#  rights in this software.
+#  This software is distributed under the 3-clause BSD License.
+#  ___________________________________________________________________________
+
+
+class Factory(object):
+    """
+    A class that is used to define a factory for objects.
+
+    Factory objects may be cached for future use.
+    """
+
+    def __init__(self, description=None):
+        self._description = description
+        self._cls = {}
+        self._doc = {}
+        self._cached = False
+        self._cache = {}
```

Comment:
Since we are not bringing the `CachedFactory` over, we can delete `_cached` and `_cache` (and all code that triggers off of them)